evm/vm_state.py

```diff
@@ -141,36 +134,19 @@ def is_key_exists(self, key):
         return self._chaindb.exists(key)
 
     #
-    # Access self.prev_headers (Read-only)
+    # Access self.prev_hashes (Read-only)
     #
-    @property
-    def parent_header(self):
-        return self.prev_headers[0]
-
     def get_ancestor_hash(self, block_number):
         """
         Return the hash of the ancestor with the given block number.
         """
-        ancestor_depth = self.block_header.block_number - block_number - 1
+        ancestor_depth = self.block_number - block_number - 1
         if (ancestor_depth >= MAX_PREV_HEADER_DEPTH or
                 ancestor_depth < 0 or
-                ancestor_depth >= len(self.prev_headers)):
+                ancestor_depth >= len(self.execution_context.prev_hashes)):
```

Comment:
We could improve readability here by pulling the full contents of the `if` statement and storing in a variable and then just using the variable in the `if` statement.

```python
is_ancestor_depth_out_of_range = (
    ancestor_depth >= MAX_PREV_HEADER_DEPTH or
    ancestor_depth < 0 or
    ancestor_depth >= len(self.execution_context.prev_hashes)
))
```